kdump-switch/cos-disable-kdump.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Deploy this DaemonSet to disable hyper-threading on the nodes with the
+# "cloud.google.com/gke-smt-disabled=true" label.
+#
+# WARNING: Disabling hyper-threading might have severe performance impact on
+# your clusters and application. Please ensure that this is acceptable before
+# deploying this to your production clusters.
+#
+# WARNING: Disabling hyper-threading requires node reboot. Therefore, in order
+# to avoid disrupting your workloads, it is recommended to create a new node
+# pool with the "cloud.google.com/gke-smt-disabled=true" label in your cluster,
+# deploy the DaemonSet to disable hyper-threading in that node pool, and then
+# migrate your workloads to the new node pool.
+
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+    name: kdump-switch
+spec:
+  template:
+    metadata:
+      labels:
+        name: kdump-switch
+    spec:
+      hostNetwork: true
+      hostPID: true
+      volumes:
+      - name: kdump-helper
+        hostPath:
+          path: /usr/sbin/kdump_helper
+      - name: dev
+        hostPath:
+          path: /dev
+      - name: os-release
+        hostPath:
+          path: /etc/os-release
+      - name: kexec-crash-size
+        hostPath:
+          path: /sys/kernel/kexec_crash_size
+      - name: kexec-crash-loaded
+        hostPath:
+          path: /sys/kernel/kexec_crash_loaded
+      - name: sysrq
+        hostPath:
+          path: /proc/sysrq-trigger
+      containers:
+      - image: gcr.io/xueweiz-experimental/cos-kdump:latest
+        command: ["/bin/sh", "-c"]
+        args: ["usr/bin/kdump.sh disable && sleep infinity"]
```

Comment:
Run kdump in init container and a pause container as the regular container. No need to sleep forever here.

Reply:
Sounds good. Thanks!